test/ConfigManagerTest.cpp

```diff
@@ -0,0 +1,347 @@
+  An empty vector doesn't work with that.
+  */
+  ASSERT_THROW(config.createConfigOption<int>(
+      ad_utility::ConfigManager::VectorOfKeysForJson{}, "", &notUsed, 42);
+               , ad_utility::EmptyVectorException);
+
+  /*
+  Trying to add a configuration option with a path containing strings with
+  spaces should cause an error.
+  Reason: A string with spaces in it, can't be read by the short hand
+  configuration grammar. Ergo, you can't set values, with such paths per short
+  hand, which we don't want.
+  */
+  ASSERT_THROW(config.createConfigOption<int>(
+      ad_utility::ConfigManager::VectorOfKeysForJson{"Shared part",
+                                                     "Sense_of_existence"},
+      "", &notUsed, 42);
+               , ad_utility::NotValidShortHandNameException);
+}
+
+TEST(ConfigManagerTest, ParseConfig) {
+  ad_utility::ConfigManager config{};
+
+  // Adding the options.
+  int firstInt;
+  int secondInt;
+  int thirdInt;
+
+  config.createConfigOption<int>(
+      ad_utility::ConfigManager::VectorOfKeysForJson{"depth_0", "Option_0"},
+      "Must be set. Has no default value.", &firstInt);
+  config.createConfigOption<int>({"depth_0", "depth_1", "Option_1"},
+                                 "Must be set. Has no default value.",
+                                 &secondInt);
+  config.createConfigOption<int>("Option_2", "Has a default value.", &thirdInt,
+                                 2);
+
+  // For easier access to the options.
+  auto getOption = [&config](const size_t& optionNumber) {
+    if (optionNumber == 0) {
+      return config.getConfigurationOptionByNestedKeys({"depth_0", "Option_0"});
+    } else if (optionNumber == 1) {
+      return config.getConfigurationOptionByNestedKeys(
+          {"depth_0", "depth_1", "Option_1"});
+    } else {
+      return config.getConfigurationOptionByNestedKeys({"Option_2"});
+    }
+  };
+
+  // For easier checking.
+  auto checkOption = [](const ad_utility::ConfigOption& option,
+                        const auto& content) {
+    ASSERT_TRUE(option.hasSetDereferencedVariablePointer());
+    ASSERT_EQ(content, option.getValue<std::decay_t<decltype(content)>>());
+  };
+
+  // Does the option with the default already have a value?
+  checkOption(getOption(2), 2);
+
+  // The other two should never have set the variable, that the internal pointer
+  // points to.
+  ASSERT_FALSE(getOption(0).hasSetDereferencedVariablePointer());
+  ASSERT_FALSE(getOption(1).hasSetDereferencedVariablePointer());
+
+  // The json for testing `parseConfig`. Sets all of the configuration
+  // options.
+  const nlohmann::json testJson(nlohmann::json::parse(R"--({
+"depth_0": {
+  "Option_0": 10,
+  "depth_1": {
+    "Option_1": 11
+  }
+},
+"Option_2": 12
+})--"));
+
+  // Set and check.
+  config.parseConfig(testJson);
+
+  checkOption(getOption(0), 10);
+  checkOption(getOption(1), 11);
+  checkOption(getOption(2), 12);
```

Comment:
You definitely have to check, that the three ints above (`firstInt` etc.) are also set to the correct values, that is the most important thing about these config options:)

Reply:
Ah, but I do! `getValue` returns the de-referenced value of the internal variable pointer. So it's the same effect, but I don't have to pass a bunch of variables too!

Reply:
Sttill. You want to document this function, and you want that after the `parse` all the external variables have the same value. If you just test a `getValue` function, then it still might be something fancy going on inside.
I explicitly want my `int` that I passed in to be changed by `parse`, so I insist on it:)

Reply:
Done, see [c7013fa](https://github.com/ad-freiburg/qlever/pull/977/commits/c7013fafa60f33ccff75a739a51ce993f8b1fa43).